tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Grpc/CallOptionsStruct.cs

```diff
@@ -0,0 +1,21 @@
+﻿// <copyright file="CallOptionsStruct.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#nullable enable
+
+using Datadog.Trace.DuckTyping;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.Grpc
+{
+    /// <summary>
+    /// Duck type for CallOptions
+    /// https://github.com/grpc/grpc/blob/master/src/csharp/Grpc.Core.Api/CallOptions.cs
+    /// </summary>
+    [DuckCopy]
+    internal struct CallOptionsStruct
```

Comment:
not sure why this is defined here, where it's only seems to be used `Datadog.Trace.ClrProfiler.AutoInstrumentation.Grpc.GrpcDotNet.GrpcNetClient.CallOptionsStruct`

Reply:
It's also used in https://github.com/DataDog/dd-trace-dotnet/pull/2545, so it makes sense to be here in the context of _both_ PRs (though I agree, not necessarily for this one alone). In my various refactoring it seemed easier to keep it in the same final location 🙂